python/py-buildbot/Portfile

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-buildbot
```

Comment:
It should not get the `py-*` prefix, which is meant for python modules. As this is software that is meant to be used as a command line utility or daemon, just give it the name `buildbot`.

Actually, the [existing buildbot port](https://github.com/macports/macports-ports/blob/6c245b1b323cadd75feb39ec82c4bb43d9b155bb/devel/buildbot/Portfile) is a good example how it should be done. :-)

Depending on the content, this might also apply to py-buildbot-pkg, if it is not a pure Python module. I have not checked what it provides.

Reply:
A while back I checked (I'll do it again now) and most ports basically install everything to the python framework, so in a way I would be in favour of keeping it this way, and then maybe provide an **additional** port `buildbot` which simply depends on `py37-buildbot`, makes sure that we get a proper command without the python version suffix, and provides all the other specialities that we currently have in that port.

So: I would not rush away from this solution just yet, and let's discuss it first. FreeBSD also does it in exact same way.

Reply:
FreeBSD might do it this way, most Linux distributions will not. In MacPorts we always tried to move away from the `py-*` prefix if the port provides a command line tool with the same name.

As a user, I want to get the `buildbot` command, so I install the port with the same name. That it is written in Python is an implementation detail I do not care about.

Having another port with a non-prefixed name sounds like a good compromise. Although it also pins it to a specific python version, unless we also provide +pythonXY variants on it. To avoid version bumps in lots of files, maybe that could be implemented as yet another subport?

Reply:
> As a user, I want to get the buildbot command, so I install the port with the same name. That it is written in Python is an implementation detail I do not care about.

Even with the `py-*` port, we do get the `buildbot` command (to be precise, I'm getting `buildbot-3.7` command)

Reply:
Here's my 2 cents:
* I definitely agree that at the end we want to end up with a port called just `buildbot`. However there are two approaches: we either put everything there, or we create a "meta-port" which pulls in all the dependencies (py37-buildbot, py37-buildbot-pkg, py37-buildbot-www, py37-buildbot-waterfall-view, ...), prepares the template files for launchctl, adds ports notes and creates symlink `buildbot` -> `buildbot-3.7`.
* Creating subports for dependencies could be an option, but this means that we loose the ability to automatically update version and dependencies etc. in case `upt` matures enough to be able to do that.
* I would like to avoid spending a lot of time nitpicking about the superior idea of how to simplify or complicate life by creating a different port organisation, and in particular I would prefer if Rajdeep spends time writing the tricky JS code rather than fighting with tricky portfiles.
* If we keep the ports as `py-buildbot` for now, we don't need to immediately take any actions to handle conflicts and renames of the old buildbot, as none of the ports currently conflict with each other.

I would go with the easy way for now and try to make sure that we can get a working set of buildbot ports, and then be able to set up a test deployment version using those. Once we have this working, we can spend an arbitrary amount of time fine-tuning port layout ourselves.

Reply:
Okay, if we want to focus, I would prefer to follow the approach that is in use for other ports (such as the existing buildbot port) instead of inventing an alternative.

Having both `buildbot @0.8.12` and `py-buildbot @2.3.1` would be confusing. We should move `buildbot` to `buildbot-0.8` (or `*-legacy` or similar) in this process and mark them as conflicting.

Reply:
@raimue @ryandesign: What do you think about the following approach?
* The worker gets installed as `buildbot-worker` and works basically in the same way as the old slave. The old `buildbot-slave` already moved to `buildbot-slave-0.8` in the meantime. I guess we all agree about that.
* I would create a port `buildbot` for the main port, but note that users might be caught by surprize to see their own port transitioned to 2.3 and suddenly broken. So my suggestion would be to temporarily create a port `buildbot-2` (so we would have `buildbot-0.8` and `buildbot-2`; I don't want to go for `buildbot-2.3` as that number is a moving target) and after a few months or a year replace it by `buildbot`, when users have had sufficient time to decide between 0.8 and 2.x. (I'm not strongly opposing just putting 2.3 into `buildbot`, it's just that some users might be caught by surprize.)
* Install dependencies (`buildbot-waterfall-view` etc.) as python modules: for now. I can turn them into regular ports later, potentially into subports to keep the same version together etc.
* This may sound a bit controversial, but at least for the time being, just make `py-buildbot` install that pypi module and nothing else. The port `buildbot-2` on the other hand would install all the dependencies (py-buildbot, py-buildbot-www, py-buildbot-pkg, py-buildbot-waterfall-view, py-buildbot-console-view, py-buildbot-grid-view, the new views to be created by @rajdeepbharati), including adding a symlink and all the trickery with creating usernames, preparing a job etc. (The exact same thing as is done for `buildbot-0.8`, just not compiling the files on its own.) I'm still confused about how alternatives could